TACC/Core-Portal
# FP-1677: Update (Fix) Button Styles

## Overview: ##

Fix button component styles.

## Related Jira tickets: ##

* [FP-1677](https://jira.tacc.utexas.edu/browse/FP-1677)
* mirrored by https://github.com/TACC/Core-Styles/pull/...
* mirrored by https://github.com/TACC/tup-ui/pull/...

## Summary of Changes: ##

- Add tertiary button section in UI Patterns.
- Show all button types loading state in UI Patterns.
- Add tertiary button styles and prop.

## UI Photos:

<details open><summary><strong>After</strong> Merge of <code>main</code> Since #639</summary>

| types | before busy | during busy |
| - | - | - |
| ![Screen Shot 2022-06-02 at 17 43 45](https://user-images.githubusercontent.com/62723358/171751111-65ce8e14-492c-48f5-b843-6bffe6c1376e.png) | ![Screen Shot 2022-06-02 at 17 44 00](https://user-images.githubusercontent.com/62723358/171751110-4162ca6c-afd8-440d-8718-0bebed5928aa.png) | ![Screen Shot 2022-06-02 at 17 44 04](https://user-images.githubusercontent.com/62723358/171751109-865f8e14-96f0-4447-9c3c-f694bd1b65ad.png) |

https://user-images.githubusercontent.com/62723358/171751105-83b64923-e7d5-4ba9-9baf-cb237cfe60b2.mov

</details>
<details><summary><strong>Before</strong> Merge of <code>main</code> Since #639</summary>

| types | before busy | during busy |
| - | - | - |
| <img width="460" alt="FP-1677 Button Types" src="https://user-images.githubusercontent.com/62723358/171696114-3d85a325-843d-4bc6-b16c-ad2bf8e3d9d7.png"> | <img width="435" alt="FP-1677 Before Busy State" src="https://user-images.githubusercontent.com/62723358/171696113-23a7d339-2fa0-4ab8-9a64-2933b888f406.png"> | <img width="435" alt="FP-1677 During Busy State" src="https://user-images.githubusercontent.com/62723358/171696108-ca7056bf-f8bb-45c6-8b4c-bfec81a63705.png"> |

https://user-images.githubusercontent.com/62723358/171697382-009230cb-699c-4c86-bb2d-67238fa6e202.mov

</details>

## Testing Steps: ##

1. Visit UI Patterns.
2. Scroll to Buttons.
3. Verify **secondary** button states matches "On Light Portal" in [design](https://xd.adobe.com/view/b37b291a-7798-4814-90f3-7ac63c48a3e0-e628/screen/d7a72466-4ae8-43fd-9531-44e033245ddb/specs/).
3. Verify **tertiary** button states matches _secondary_ button states from #598.
4. Verify busy state ("Spinner over Button") is distinct for each button type.
